flutter/lib/ui/history/result_filter_screen.dart

```diff
@@ -93,30 +97,43 @@ class _ResultFilterScreenState extends State<ResultFilterScreen> {
   Widget _platformFilter(ResultFilter filter) {
     return _makeDropDownFilter(
         labelText: l10n.historyFilterPlatform,
-        choices: EnvPlatform.values.map((e) => e.name).toList(),
-        value: filter.platform,
-        onChanged: (value) => setState(() {
-              filter.platform = value!;
+        choices: EnvPlatform.values
+            .map((e) => DropdownOption(e.name, e.name))
+            .toList(),
+        value: filter.platform != null
+            ? DropdownOption(filter.platform, filter.platform ?? '')
+            : null,
+        onChanged: (option) => setState(() {
+              filter.platform = option?.value as String;
```

Comment:
The app crashed when setting platform filter and sorting by task throughput.
```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following _CastError was thrown building ResultListScreen(dirty, dependencies:
[_InheritedProviderScope<BenchmarkState?>, _LocalizationsScope-[GlobalKey#ae549]], state:
_ResultListScreenState#353e7):
Null check operator used on a null value

The relevant error-causing widget was:
  ResultListScreen
  ResultListScreen:file:///Users/anh/dev/mlcommons/mobile_app_open/flutter/lib/ui/run/app_bar.dart:20:59

When the exception was thrown, this was the stack:
#0      ResultSort.sortBenchmarks.<anonymous closure> (package:mlperfbench_common/data/result_sort.dart:32:46)
#1      Sort._insertionSort (dart:_internal/sort.dart:69:36)
#2      Sort._doSort (dart:_internal/sort.dart:58:7)
#3      Sort.sort (dart:_internal/sort.dart:33:5)
#4      ListMixin.sort (dart:collection/list.dart:356:10)
#5      ResultSort.sortBenchmarks (package:mlperfbench_common/data/result_sort.dart:32:15)
#6      ResultsDataProvider.resultItems (package:mlperfbench/ui/history/results_data_provider.dart:26:25)
#7      _ResultListScreenState.build (package:mlperfbench/ui/history/result_list_screen.dart:35:56)
#8      StatefulElement.build (package:flutter/src/widgets/framework.dart:5080:27)
#9      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4968:15)
#10     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:5133:11)
#11     Element.rebuild (package:flutter/src/widgets/framework.dart:4690:5)
#12     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2743:19)
#13     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:863:21)
#14     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:381:5)
#15     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1289:15)
#16     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1218:9)
#17     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1076:5)
#18     _invoke (dart:ui/hooks.dart:145:13)
#19     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:338:5)
#20     _drawFrame (dart:ui/hooks.dart:112:31)

════════════════════════════════════════════════════════════════════════════════════════════════════
```

Reply:
Can't reproduce anymore